src/st_fns.py

```diff
@@ -36,62 +35,131 @@ def check_un(engine, user_name, new_or_exist):
     
+    # Check if search query is not empty
+    if search_query:
+        if 10 >= num_results > 1:
+            st.write("More than one result found, please narrow down search query from options below")
+            st.write(df_search)
+        elif num_results == 0:
+            st.write("No results found")
+        elif num_results > 10:
+            st.write("Too many results found to display")
+        else:
+            selected_key = df_search["film_key"].values[0]
+            search_film_desc_w = st.empty()
+            
+            # Film info to be stored with rating
+            search_current_film_key = selected_key
+        
+            # Save space to display film info
+            search_film_info_w = st.empty()
+            search_film_desc_w = st.empty()
+            
+            search_film_title = df_search["title"].values[0]
+            search_film_year = int(df_search["year"].values[0])
+            search_film_description = df_search["description"].values[0]
+            
+            search_film_info_w.write(f'Do you want to watch {search_film_title}, released in {search_film_year}')
+            search_film_desc_w.write(search_film_description)
+            
+            # Text rating is mapped to number
+            # Only stored once user clicks a button
+            rating = -99
+            if st.button('Yes, looks rad!', key="search_good"):
+                rating = 5
+            if st.button('No, I have taste!', key="search_bad"):
+                rating = 0
+            if st.button('Skip', key="search_skip"):
+                rating = -1
+            if rating != -99:
+                # Create new hash for new random film
+                dict_cache['random_hash'] += random.randrange(-n_films, n_films)
+                add_user_rating(dict_cache['engine'], dict_cache["user_name"], search_current_film_key, rating)
+               
+            # Clear search once search has found a film
+            search_query = search_placeholder.text_input('Film name', value='', key=1)
+        
+    else:
+        st.write("Please enter film to search for")
+        
+   
     st.header('Rate Films')
     st.write('Please select your preferences for movies below')
     
     # Save space to display film info
     film_info_w = st.empty()
     film_desc_w = st.empty()
     
+    # Get films rated by group, but not rated by user
+    n_films = get_film_count(dict_cache['engine'])
+    urated_films_df = get_user_rated_films(dict_cache["engine"], dict_cache["user_name"])
+    grated_films_df = get_group_rated_films(dict_cache["engine"], dict_cache["group_name"])
+    
+    df_merge = urated_films_df.merge(grated_films_df, on='film_key', how='outer', indicator=True)
+    films_to_rate = df_merge.loc[df_merge['_merge'] == "right_only"]["film_key"].tolist()
+    print(films_to_rate)
+    if films_to_rate:
+        rand_film = random.choice(films_to_rate)
+        films_to_rate.remove(rand_film)
```

Comment:
Why doesn't this list get updated when the code just above gets rerun? I didn't think I'd have to edit the list explicitly